src/core/libmaven/isotopeDetection.cpp

```diff
@@ -164,58 +152,83 @@ map<string, PeakGroup> IsotopeDetection::getIsotopes(PeakGroup* parentgroup, vec
+            return true;
+        float observedAbundance = isotopePeakIntensity
+            / (parentPeakIntensity + isotopePeakIntensity); //find observedAbundance based on isotopePeakIntensity
+
+        float naturalAbundanceError = abs(
+                observedAbundance - expectedAbundance) //if observedAbundance is significant wrt expectedAbundance
+            / expectedAbundance * 100; // compute natural Abundance Error
+
+        if (naturalAbundanceError >
+                _mavenParameters->maxNaturalAbundanceErr)
+            return true;
+    }
+
+    //TODO: this is really an abuse of the maxIsotopeScanDiff parameter
+    //I can easily imagine you might set maxIsotopeScanDiff to something much less than the peak width
+    //here w should really be determined by the minRt and maxRt for the parent and child peaks
+    if (parentGroup)
     {
-        //TODO: this is really an abuse of the maxIsotopeScanDiff parameter
-        //I can easily imagine you might set maxIsotopeScanDiff to something much less than the peak width
-        //here w should really be determined by the minRt and maxRt for the parent and child peaks
-        float w = _mavenParameters->maxIsotopeScanDiff
-            * _mavenParameters->avgScanTime;
-        float isotopeMass = x.mass;
         Peak* parentPeak = parentGroup->getPeak(sample);
-        float rtmin = parentPeak->rtmin;
-        float rtmax = parentPeak->rtmax;
+        float rtmin = parentGroup->minRt;
+        float rtmax = parentGroup->maxRt;
+        if (parentPeak)
+        {
+            rtmin = parentPeak->rtmin;
+            rtmax = parentPeak->rtmax;
+        }
+        float isotopeMass = x.mass;
         float parentMass = parentGroup->meanMz;
+        float w = _mavenParameters->maxIsotopeScanDiff
+            * _mavenParameters->avgScanTime;
         double c = sample->correlation(
                 isotopeMass, parentMass,
                 _mavenParameters->compoundMassCutoffWindow, rtmin - w,
                 rtmax + w, _mavenParameters->eicType,
-                _mavenParameters->filterline);  // find correlation for isotopes
+            _mavenParameters->filterline);  // find correlation for isotopes
         if (c < _mavenParameters->minIsotopicCorrelation)
             return true;
     }
     return false;
 }
 
+std::pair<float, float> IsotopeDetection::getIntensity(Scan* scan, float mzmin, float mzmax)
+{
+    float highestIntensity = 0;
+    float rt = 0;
+    mzSample* sample = scan->getSample();
+    //TODO: use maxIsotopeScanDiff instead of arbitrary number
+    for (int i = scan->scannum - 2; i < scan->scannum + 2; i++) {
+		Scan* s = sample->getScan(i);
+		vector<int> matches = s->findMatchingMzs(mzmin, mzmax);
+		for (unsigned int i = 0; i < matches.size(); i++) {
```

Comment:
@shubhra-agrawal 
just a suggestion. we can use [range based for](http://en.cppreference.com/w/cpp/language/range-for) here

Reply:
Yep. Makes sense. I'll add it.